src/track/beatgrid.cpp

```diff
@@ -147,9 +147,27 @@ audio::FramePos BeatGrid::findNthBeat(audio::FramePos position, int n) const {
         return audio::kInvalidFramePos;
     }
 
-    const double beatFraction = (position - firstBeatPosition()) / m_beatLengthFrames;
-    const double prevBeat = floor(beatFraction);
-    const double nextBeat = ceil(beatFraction);
+    double beatFraction = (position - firstBeatPosition()) / m_beatLengthFrames;
+    double prevBeat = floor(beatFraction);
+    double nextBeat = ceil(beatFraction);
+
+    // If the position is within 1/100th of the next or previous beat, treat it
+    // as if it is that beat.
+    const double kEpsilon = .01;
```

Comment:
This should be passed as a parameter instead if hardcoding it, e.g. `beatPositionTolerance`. A value of 0.0 should disable the *fuzzy* behavior. The caller knows best about the precision of the `position`.

Reply:
A 1/100th of a frame?

Reply:
@Holzhaus Other ideas?

Reply:
What would be a use case for setting a different value for this?

Reply:
Counter question: What is the rationale for the hard-coded constant `0.01`?

Reply:
I presume it was determined experimentally in the old code this is reverting to.